src/index.ts

```
@@ -21,6 +22,89 @@ function writeLoaders(args: CLIArgs) {
 }
 
+/**
+ * If requiredList only contains responseKey
+ */
+function validRequiredList(requiredList: Array<string>, responseKey: string) {
+    if (requiredList.length > 1) return false;
+    if (requiredList.length == 1 && requiredList[0] !== responseKey) {
+        return false;
+    }
+    return true;
+}
+
+/**
+ * Find all values of the intput key in an object recursively
+ */
+function findVal(object: any, key: string, array: Array<string>) {
+    Object.keys(object).some(function (k) {
+        if (k === key) {
+            array.push(...object[k]);
+        }
+        if (object[k] && typeof object[k] === 'object') {
+            findVal(object[k], key, array);
+        } else {
+            return;
+        }
+    });
+    return array;
+}
+
+/**
+ * Throw erros when resource uses propertyBatchKey but doesn't match requirements
+ */
+function verifyBatchPropertyResource(args: CLIArgs) {
+    const resources: object = getConfig(args.config).resources;
+
+    Object.entries(resources).forEach(([key, value]) => {
+        if (Object.keys(value).includes('propertyBatchKey')) {
+            const propertyBatchKey = value['propertyBatchKey'];
+            if (
+                !value.hasOwnProperty('responseKey') ||
+                !value.hasOwnProperty('swaggerLink') ||
+                !value.hasOwnProperty('swaggerPath') ||
+                !value.hasOwnProperty('httpMethod')
+            ) {
+                throw new Error(`Missing swagger info for ${key}, please add them in the dataloader-cofig.yaml file!`);
+            }
+            const swaggerLink: string = value['swaggerLink'];
+            const swaggerPath: string = value['swaggerPath'];
+            const httpMethod: string = value['httpMethod'];
+
+            // parse swagger file, so that all $ref pointers will be resolved.
+            SwaggerParser.validate(swaggerLink, (err, api) => {
+                if (err) {
+                    console.error(err);
```

is silently failing a good thing? do we have cases where the swagger spec fails to parse but "still works"? or can we hard fail here?